File: proxy/tests/script_engine/url_host.js
/// <reference path="../../script_libs/js/index.d.ts" />
/** @type {Extension} */
const url_host = {
  request(flow) {
    if (flow.request.url.host = "localhost:1234") {
      flow.request.url.host = "example.com:4321"
    }
  },
};
globalThis.extensions = [url_host]

File: proxy/tests/script_engine/url_host.lua
Extensions = {
	{
		request = function(flow)
			print("Original host: " .. flow.request.url.host)
			if flow.request.url.host == "localhost:1234" then
				flow.request.url.host = "example.com:4321"
			end
		end,
	},
}
